packages/gestalt/src/Masonry/defaultLayout.js

```diff
@@ -10,19 +11,51 @@ const offscreen = (width: number, height: number = Infinity) => ({
   height,
 });
 
+const calculateCenterOffset = ({
+  columnCount,
+  columnWidthAndGutter,
+  gutter,
+  justify,
+  layout,
+  rawItemCount,
+  width,
+}: {
+  columnCount: number,
+  columnWidthAndGutter: number,
+  gutter: number,
+  justify: Align,
+  layout: Layout,
+  rawItemCount: number,
+  width: number,
+}): number => {
+  if (layout === 'basicCentered') {
+    const contentWidth = Math.min(rawItemCount, columnCount) * columnWidthAndGutter + gutter;
+    return Math.max(Math.floor((width - contentWidth) / 2), 0);
+  }
+  if (justify === 'center') {
+    return Math.max(Math.floor((width - columnWidthAndGutter * columnCount + gutter) / 2), 0);
+  }
+  if (justify === 'end') {
+    return width - (columnWidthAndGutter * columnCount - gutter);
+  }
+  return 0;
+};
+
 const defaultLayout =
   <T>({
+    justify,
     cache,
     columnWidth = 236,
     gutter = 14,
-    justify,
+    layout,
     minCols = 2,
     rawItemCount,
     width,
   }: {
     columnWidth?: number,
     gutter?: number,
-    justify: 'center' | 'start',
+    justify: Align,
```

Comment:
Let me know if this isn't okay, felt like we could just keep this prop as `justify`. @liuyenwei

Reply:
let's keep them consistent